src/byte_format/deserialize.cr

```diff
@@ -67,7 +67,11 @@ struct Crystalizer::ByteFormat
   end
 
   def deserialize(to type : Number::Primitive.class)
-    @io.read_bytes type, @format
+    begin
+      @io.read_bytes type, @format
+    rescue e
+      raise Error.new e.message
+    end
```

Comment:
Why this change?

Reply:
It seemed to me that you wanted to always return a local Error.new from (de)serializer modules, and without this the user would see the underlying IO::Error instead of some local Error. Do you want to keep or undo this?

Reply:
This means doing this for all other methods too.
We may consider this in another PR.

Reply:
Also here the `cause` was missing.

Reply:
Ah yes, actually this was an unintended change, I intended to do this in another place, not for Number. Undo-ed it now, thanks.